src/store/modules/blocks/actions.js

```diff
@@ -15,36 +18,105 @@ const { startLoading, endLoading } = createActionHelpers({
  */
 export default {
   /**
-   * height fetches the blockheight
+   * height fetches the block-height
    * @param {Object} state
    * @param {Function} commit
    * @param {Function} dispatch
    * @return {Object}
    */
   async height ({ state, commit, dispatch }) {
-    /**
-     * Start vuex-loading
-     */
     startLoading(dispatch, 'blocks/height')
 
-    /**
-     * Fetch the top block
-     */
-    const block = await fetch(`${process.env.AETERNITY_EPOCH_API_URL}v2/top`)
+    const client = await ae
+    const height = await client.height()
 
-    /**
-     * Get the hight of the response
-     */
-    const { height } = await block.json()
+    if (height === state.height) {
+      endLoading(dispatch, 'blocks/height')
+      return state.height
+    }
 
-    /**
-     * Start vuex-loading
-     */
+    commit('setHeight', height)
     endLoading(dispatch, 'blocks/height')
 
-    /**
-     * Commit the updates
-     */
-    return commit('setHeight', height)
+    return height
+  },
+
+  /**
+   * getBlockFromHash fetches the block from the blockchain
+   * from a single hash argument
+   * @param {Object} state
+   * @param {Function} commit
+   * @param {Function} dispatch
+   * @param {String} hash
+   * @return {*}
+   */
+  async getBlockFromHash ({ state, commit, dispatch }, hash) {
+    startLoading(dispatch, 'blocks/getBlockFromHash')
+
+    const client = await ae
+    const block = await client.api.getBlockByHash(hash, { txEncoding: 'json' })
+
+    if (isEqual(state.block, block)) {
+      endLoading(dispatch, 'blocks/getBlockFromHash')
+      return state.block
+    }
+
+    commit('setBlock', block)
+    endLoading(dispatch, 'blocks/getBlockFromHash')
+
+    return block
+  },
+
+  /**
+   * Fetches the head of the blockchain
+   * @param {Object} state
+   * @param {Function} commit
+   * @param {Function} dispatch
+   * @param {String} height
+   * @return {*}
+   */
+  async getBlockFromHeight ({ state, commit, dispatch }, height) {
+    startLoading(dispatch, 'blocks/getBlockFromHeight')
+
+    const client = await ae
+    const block = await client.api.getBlockByHeight(height, { txEncoding: 'json' })
+
+    if (isEqual(state.block, block)) {
+      endLoading(dispatch, 'blocks/getBlockFromHeight')
+      return state.block
+    }
+
+    commit('setBlock', block)
+    endLoading(dispatch, 'blocks/getBlockFromHeight')
```

Comment:
I think `startLoading(dispatch, <method name>)` and `endLoading(dispatch, <method name>)` should be extracted to decorator to avoid duplication of this pattern.